Jenkinsfile

```diff
@@ -0,0 +1,106 @@
+@Library('jenkins.shared.library') _
+
+pipeline {
+  agent {
+    label 'ubuntu_docker_label'
+  }
+  tools {
+    go "Go 1.13"
+  }
+  options {
+    checkoutToSubdirectory('src/github.com/infobloxopen/ingress-nginx')
+  }
+  environment {
+    DIRECTORY = "src/github.com/infobloxopen/ingress-nginx"
+    GIT_VERSION = sh(script: "cd ${DIRECTORY} && git describe --always --long --tags",
+                       returnStdout: true).trim()
+    TAG = "${env.GIT_VERSION}-j${env.BUILD_NUMBER}"
+    REGISTRY = 'infoblox'
+    IMGNAME = 'nginx-fips'
+    PLATFORMS = 'amd64'
+    ARCH = 'amd64'
+    MULTI_ARCH_IMG="${env.REGISTRY}/${env.IMGNAME}-${env.ARCH}"
+    BASEIMAGE = "${REGISTRY}/${env.IMGNAME}:${TAG}"
+    DOCKER_CLI_EXPERIMENTAL = 'enabled'
+    GOPATH = "$WORKSPACE"
+  }
+  stages {
+    stage("Setup docker-ce") {
+        steps {
+          sh "curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -"
+          sh 'sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"'
+          sh "sudo apt-get update"
+          sh 'sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce'
```

Comment:
The purpose of labeling nodes `ubuntu_docker_label` is to indicate they already have docker installed. What necessitates reinstalling it?

Reply:
> The purpose of labeling nodes `ubuntu_docker_label` is to indicate they already have docker installed. What necessitates reinstalling it?

Thanks, for the clarification but the installed docker doesn't support `DOCKER_CLI_EXPERIMENTAL` functionality which is required by the build scripts, exempli gratia: https://github.com/infobloxopen/ingress-nginx/blob/40678ed77a10bccc1e273ce2084109b1cdff8264/images/nginx/Makefile#L31.
I'm not very experienced with Jenkins, so if you know how to enable this functionality in shipped docker, please provide it.

Reply:
@alavrovinfb Looks like the preinstalled version on `ubuntu_docker_label` is 18.09.7. According to https://docs.docker.com/buildx/working-with-buildx/, "Docker Buildx is included in Docker 19.03" 
It would be better to provision a new node (or update the existing config for `ubuntu_docker_label`) to use a different version. This will require a devops ticket.
In the mean time , it looks like there are other build issues to resolve? https://jenkins-cd.inca.infoblox.com/job/infobloxopen/job/ingress-nginx/job/ATLAS-8981-test/54/console

Reply:
> @alavrovinfb Looks like the preinstalled version on ubuntu_docker_label is 18.09.7. According to https://docs.docker.com/buildx/working-with-buildx/, "Docker Buildx is included in Docker 19.03"
It would be better to provision a new node (or update the existing config for ubuntu_docker_label) to use a different version. This will require a devops ticket.

I've fixed this by installing the latest docker-ce version. I'm Ok with my solution.

> In the mean time , it looks like there are other build issues to resolve? https://jenkins-cd.inca.infoblox.com/job/infobloxopen/job/ingress-nginx/job/ATLAS-8981-test/54/console

Yes, there is a minor issue with two unit test caused by this `listen tcp :10246: bind: address already in use` error. It is strange because this issue doesn't reproduce either in my local environment and in the TravisCi as well. I think it somehow depends on our Jenkins build system.  Anyway, it doesn't seem to influent image build but to be sure these tests should be fixed.

```
--- FAIL: TestConfigureDynamically (0.00s)
    nginx_test.go:154: crating tcp listener: listen tcp :10246: bind: address already in use
=== RUN   TestConfigureCertificates
--- FAIL: TestConfigureCertificates (0.00s)
    nginx_test.go:323: crating tcp listener: listen tcp :10246: bind: address already in use
```
Also, there is an issue with permissions, images can't be pushed to the docker hub due to access denied. I've mentioned this issue in the Jira tickets.

Reply:
Maybe you could add a conditional section around this stage that checks for the availability of the feature you need, so that it only reinstalls docker when the docker version is unsupported. That way once the nodes are upgraded it will skip the stage.

Reply:
Done